discord/ext/menus/pagination.py

```diff
@@ -0,0 +1,264 @@
+from typing import List, Union
+
+import discord
+from discord import abc
+from discord.commands import ApplicationContext
+from discord.ext.commands import Context
+
+
+class PaginateButton(discord.ui.Button):
+    def __init__(self, label, emoji, style, disabled, button_type, paginator):
+        super().__init__(label=label, emoji=emoji, style=style, disabled=disabled, row=0)
+        self.label = label
+        self.emoji = emoji
+        self.style = style
+        self.disabled = disabled
+        self.button_type = button_type
+        self.paginator = paginator
+
+    async def callback(self, interaction: discord.Interaction):
+        if self.button_type == "first":
+            self.paginator.current_page = 0
+        elif self.button_type == "prev":
+            self.paginator.current_page -= 1
+        elif self.button_type == "next":
+            self.paginator.current_page += 1
+        elif self.button_type == "last":
+            self.paginator.current_page = self.paginator.page_count
+        await self.paginator.goto_page(interaction=interaction, page_number=self.paginator.current_page)
+
+
+class Paginator(discord.ui.View):
+    """Creates a paginator for a message that is navigated with buttons.
```

Comment:
i personally don't like "navigated" used here maybe use "filled" or "implemented with buttons."

Reply:
This one is on my list to redo - still has the original verbiage from the old PR.